install-steps.md

```diff
@@ -1,193 +1,821 @@
+            hardwareProfile: default
+          - name: openshift-worker-0
+            role: worker
+            bmc:
+              address: ipmi://<out-of-band-ip>
+              username: <user>
+              password: <password>
+            bootMACAddress: <NIC1-mac-address
+            hardwareProfile: default
+          - name: openshift-worker-1
+            role: worker
+            bmc:
+              address: ipmi://<out-of-band-ip>
+              username: <user>
+              password: <password>
+            bootMACAddress: <NIC1-mac-address
+            hardwareProfile: default
+    pullSecret: '<pull_secret>'
+    sshKey: '<ssh_pub_key>'
+    ~~~
+
+NOTE: Ensure to change the appropriate variables to match your environment.
+
+2. Create a directory to store cluster configs
+    ~~~sh
+    mkdir ~/clusterconfigs
+    cp install-config.yaml ~/clusterconfigs
+    ~~~
+
+3. Ensure all baremetal nodes are powered off prior to installing the OpenShift cluster
+   ~~~sh
+   ipmitool -I lanplus -U <user> -P <password> -H <management-server-ip> -p 6230 power off
+   ~~~
+
+4. Run the OpenShift Installer
+    ~~~sh
+    ./openshift-baremetal-install --dir ~/clusterconfigs --log-level debug create cluster
+    ~~~
+
+## Fix Metal3 ConfigMap
+
+With the installation complete. The focus shifts to fixing the metal3 ConfigMap
+in order to deploy new nodes. 
+
+Due to this [issue](https://github.com/openshift/installer/pull/2449) we need 
+to fix the ConfigMap for the Metal3 operator.
+
+This ConfigMap is used to notify `ironic` how to PXE boot new nodes.
+
+1. Create the sample ConfigMap `metal3-config.yaml.sample`
+   ~~~yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: metal3-config
+      namespace: openshift-machine-api
+    data:
+      cache_url: ${RHCOS_URI}
+      deploy_kernel_url: http://172.22.0.3:6180/images/ironic-python-agent.kernel
+      deploy_ramdisk_url: http://172.22.0.3:6180/images/ironic-python-agent.initramfs
+      dhcp_range: 172.22.0.10,172.22.0.100
+      http_port: "6180"
+      ironic_endpoint: http://172.22.0.3:6385/v1/
+      ironic_inspector_endpoint: http://172.22.0.3:5050/v1/
+      provisioning_interface: <NIC1>
+      provisioning_ip: 172.22.0.3/24
+      rhcos_image_url: ${RHCOS_PATH}${RHCOS_URI}
+    ~~~
+    NOTE: The `provision_ip` should be modified to an available IP on the `provision` network. The default is `172.22.0.3`
+
+2. Create the final ConfigMap
+    ~~~sh
+    export COMMIT_ID=$(./openshift-baremetal-install version | grep '^built from commit' | awk '{print $4}')
+    export RHCOS_URI=$(curl -s -S https://raw.githubusercontent.com/openshift/installer/$COMMIT_ID/data/data/rhcos.json | jq .images.openstack.path | sed 's/"//g')
+    export RHCOS_PATH=$(curl -s -S https://raw.githubusercontent.com/openshift/installer/$COMMIT_ID/data/data/rhcos.json | jq .baseURI | sed 's/"//g')
+    envsubst < metal3-config.yaml.sample > metal3-config.yaml
```

Comment:
i would inject this configmap prior to the deployment, so no fix is required after launching the openshift.install command

Reply:
@karmab ++

Reply:
@davidvossel @karmab - today I went through this and was also trying to figure out how would you add this fix prior to launching the opneshift install command?

Reply:
I do the following:
        OPENSHIFT_INSTALLER=/usr/local/bin/openshift-install
        OPENSHIFT_INSTALL_COMMIT=$($OPENSHIFT_INSTALLER version | grep commit | cut -d' ' -f4)
        OPENSHIFT_INSTALLER_RHCOS=${OPENSHIFT_INSTALLER_RHCOS:-https://raw.githubusercontent.com/openshift/installer/$OPENSHIFT_INSTALL_COMMIT/data/data/rhcos.json}
        RHCOS_IMAGE_JSON=$(curl "${OPENSHIFT_INSTALLER_RHCOS}")
        export RHCOS_INSTALLER_IMAGE_URL=$(echo "${RHCOS_IMAGE_JSON}" | jq -r '.baseURI + .images.openstack.path')
        export RHCOS_IMAGE_URL=${RHCOS_IMAGE_URL:-${RHCOS_INSTALLER_IMAGE_URL}}
        sed -i "s#RHCOS_IMAGE_URL#${RHCOS_IMAGE_URL}#" /home/bschmaus/metal3-config.yaml
/usr/local/bin/openshift-install --dir=ocp create manifests
cp /home/bschmaus/metal3-config.yaml ocp/openshift/99_metal3-config.yaml
/usr/local/bin/openshift-install --dir=ocp --log-level debug create cluster

Reply:
@schmaustech I added this in the doc. Ty for the steps.